compiler/noirc_frontend/src/elaborator/lints.rs

```diff
@@ -109,6 +109,72 @@ pub(super) fn oracle_not_marked_unconstrained(
     }
 }
 
+/// Oracle functions cannot return more than 1 slice in their output.
+///
+/// This is currently a limitation with the AVM: to return multiple slices
+/// of unknown length, it would need to support to allocating memory for
+/// them in the call handler, and return their final address. Currently
```

Comment:
```suggestion
/// of unknown length, it would need to support allocating memory for
/// them in the call handler, and return their final address. Currently
```